app/uk/gov/hmrc/incometaxsubscriptionselfemployedfrontend/controllers/BusinessTradeNameController.scala

```diff
@@ -75,13 +77,18 @@ class BusinessTradeNameController @Inject()(mcc: MessagesControllerComponents,
             formWithErrors =>
               Future.successful(BadRequest(view(formWithErrors, id, isEditMode = isEditMode))),
             businessTradeNameData =>
-              multipleSelfEmploymentsService.saveBusinessTrade(id, businessTradeNameData).map(_ =>
-                if (isEditMode) {
-                  Redirect(uk.gov.hmrc.incometaxsubscriptionselfemployedfrontend.controllers.routes.BusinessListCYAController.show())
-                } else {
-                  Redirect(
-                    uk.gov.hmrc.incometaxsubscriptionselfemployedfrontend.controllers.routes.AddressLookupRoutingController.initialiseAddressLookupJourney(id))
+              multipleSelfEmploymentsService.saveBusinessTrade(id, businessTradeNameData).map(_ => {
+                val call = (isEditMode, isSaveAndRetrieve) match {
+                  case (true, true) => uk.gov.hmrc.incometaxsubscriptionselfemployedfrontend.controllers.routes.SelfEmployedCYAController.show(id)
+                  case (false, true) => uk.gov.hmrc.incometaxsubscriptionselfemployedfrontend.controllers.routes.AddressLookupRoutingController.initialiseAddressLookupJourney(id)
+                  case (true, false) => uk.gov.hmrc.incometaxsubscriptionselfemployedfrontend.controllers.routes.BusinessListCYAController.show()
+                  case (false, false) => uk.gov.hmrc.incometaxsubscriptionselfemployedfrontend.controllers.routes.AddressLookupRoutingController.initialiseAddressLookupJourney(id)
                 }
+
+                println("isEditMode: " + isEditMode)
```

Comment:
I think you forgot to remove this one :)

Reply:
sorry, :) will do it